packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -92,44 +93,77 @@ export class ModularChangeFamily
         return { fieldKind, changesets: normalizedChanges };
     }
 
-    compose(changes: FieldChangeMap[]): FieldChangeMap {
+    compose(changes: TaggedChange<FieldChangeMap>[]): FieldChangeMap {
         if (changes.length === 1) {
-            return changes[0];
+            return changes[0].change;
```

Comment:
What if the single changeset only had RevisionTag information in the `TaggedChange.revision` field, wouldn't this discard that information?
Or would the change be forced to have RevisionTag information lower down as well? If so, does this PR make the `TaggedChange.revision` field redundant?

Reply:
This early return is incorrect when the function is being called recursively. Good catch.